Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -234,98 +315,104 @@ - (BOOL)containsObject:(id)object {
+
     UNIMPLEMENTED();
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (void)enumerateObjectsUsingBlock:(void (^)(id, NSUInteger, BOOL*))block {
-    UNIMPLEMENTED();
+    return [_arrayContainer enumerateObjectsUsingBlock:block];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (void)enumerateObjectsWithOptions:(NSEnumerationOptions)opts usingBlock:(void (^)(id, NSUInteger, BOOL*))block {
-    UNIMPLEMENTED();
+    return [_arrayContainer enumerateObjectsWithOptions:opts usingBlock:block];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (id)objectAtIndex:(NSUInteger)index {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [_arrayContainer objectAtIndex:index];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
+*/
+- (id)firstObject {
+    return [_arrayContainer firstObject];
+}
+
+/**
+ @Status Interoperable
+*/
+- (id)lastObject {
+    return [_arrayContainer lastObject];
+}
+
+/**
+ @Status Interoperable
 */
 - (id)objectAtIndexedSubscript:(NSUInteger)index {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [_arrayContainer objectAtIndexedSubscript:index];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSArray*)objectsAtIndexes:(NSIndexSet*)indexes {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [_arrayContainer objectsAtIndexes:indexes];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSUInteger)indexOfObject:(id)object {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [_arrayContainer indexOfObject:object];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSUInteger)indexOfObject:(id)object
               inSortedRange:(NSRange)range
                     options:(NSBinarySearchingOptions)opts
             usingComparator:(NSComparator)cmp {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [_arrayContainer indexOfObject:object inSortedRange:range options:opts usingComparator:cmp];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSUInteger)indexOfObjectAtIndexes:(NSIndexSet*)indexSet
                              options:(NSEnumerationOptions)opts
                          passingTest:(BOOL (^)(id, NSUInteger, BOOL*))predicate {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [_arrayContainer indexOfObjectAtIndexes:indexSet options:opts passingTest:predicate];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSUInteger)indexOfObjectPassingTest:(BOOL (^)(id, NSUInteger, BOOL*))predicate {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [_arrayContainer indexOfObjectPassingTest:predicate];
 }
 
 /**
  @Status Stub
  @Notes
 */
 - (NSUInteger)indexOfObjectWithOptions:(NSEnumerationOptions)opts passingTest:(BOOL (^)(id, NSUInteger, BOOL*))predicate {
+    // TODO: uncomment when NSArray has implemented indexOfObjectWithOptions
+    // return [_arrayContainer indexOfObjectWithOptions:opts passingTest:predicate];
```

Comment:
(and everywhere else in this file)

---
In reply to: [64478371](https://github.com/Microsoft/WinObjC/pull/482#discussion_r64478371) [](ancestors = 64478371)